### PR TITLE
Use new Azure ServiceBus library that can create queues

### PR DESCRIPTION
**Pull request by DeonHeyns**
* Updated servicestack.azure.core.nuspec to use Microsoft.Azure.ServiceBus and dropped icrosoft.Azure.Management.ServiceBus
* Added a prefetch property to ServiceBusMqMessageFactory (removed, there is a separate PR for this)
* Added ctor to ServiceBusMqServer that accepts ServiceBusMqMessageFactory which allows prefetch to be configured
* Microsoft.Azure.ServiceBus library now includes management api that allows queues to be created from a connectionstring

All MQ tests run and pass (except for ignored test)

### CODE REVIEW

File: src/ServiceStack.Azure/Messaging/ServiceBusMqMessageFactory.cs [hunk 1]
```diff
@@ -124,28 +128,51 @@ protected internal void StopQueues()
...
+        {
+            var mqDesc = new QueueDescription(queueName);
+#if !NETSTANDARD2_0
+                    if (!namespaceManager.QueueExists(queueName))
+                        namespaceManager.CreateQueue(mqDesc);
+#endif
+#if NETSTANDARD2_0
+            try
+            {
+                managementClient.QueueExistsAsync(queueName)
```
Comment (DeonHeyns) [inline src/ServiceStack.Azure/Messaging/ServiceBusMqMessageFactory.cs]: @mythz wouldn't mind some guidance on how you'd like to see this part done. All management apis are async so went this route for now

File: src/ServiceStack.Azure/Messaging/ServiceBusMqMessageFactory.cs [hunk 2]
```diff
@@ -19,6 +20,9 @@ public class ServiceBusMqMessageFactory : IMessageFactory
 #if !NETSTANDARD2_0
         protected internal readonly NamespaceManager namespaceManager;
 #endif
+#if NETSTANDARD2_0
```
Comment (mythz) [inline src/ServiceStack.Azure/Messaging/ServiceBusMqMessageFactory.cs]: use `#else` instead of repeating the negative condition

**Comment by DeonHeyns** [comment]
updated all occurrences, thanks!

File: src/ServiceStack.Azure/Messaging/ServiceBusMqMessageFactory.cs (hunk 1 above)
Comment (mythz) [reply src/ServiceStack.Azure/Messaging/ServiceBusMqMessageFactory.cs]: The async method would normally be propagated back to the public API which would return a Task Response. But if you want to keep the different implementations the same as the sync API you would block and call `.Wait()` and `.Result` however this opens the chance of a deadlock.

To best mitigate the changes, ideally registering queue's would only be done at registration for all the registered queue's when the MQ is started.

**Comment by DeonHeyns** [comment]
Makes sense, I'll update the code. I'm using the RabbitMQ implementation as
a reference and was wondering when/where do the <dto>response queues get
created?

**Comment by mythz** [comment]
It's created on `Init()` called on RabbitMqServer.Start():

https://github.com/ServiceStack/ServiceStack/blob/ac714303b5e027ae59f1535de0799074ddd33622/src/ServiceStack.RabbitMq/RabbitMqServer.cs#L274

**Comment by DeonHeyns** [comment]
I’ve neglected this due to work commitments and found an issue that I’ll need to spend time resolving. So closing for now and will send a new PR.

**Pull request closed without merging**